packages/wdio-allure-reporter/src/index.ts

```diff
@@ -1,605 +1,686 @@
+            if (isSkipped) {
+                return this.allure.endCase(PENDING)
+            }
+
+            // Only close passing and skipped tests because
+            // failing tests are closed in onTestFailed event
+            return
+        }
+
+        this.allure.endSuite()
+    }
+
+    onTestStart(test: TestStats | HookStats) {
+        const testTitle = test.currentTest ? test.currentTest : test.title
+        if (this.isAnyTestRunning() && this.allure.getCurrentTest().name == testTitle) {
+            // Test already in progress, most likely started by a before each hook
+            this.setCaseParameters(test.cid)
+            return
+        }
+
+        if (this.options.useCucumberStepReporter) {
+            return this.allure.startStep(testTitle)
+        }
+
+        this.allure.startCase(testTitle)
+        this.setCaseParameters(test.cid)
+    }
+
+    setCaseParameters(cid?: string) {
+        const currentTest = this.allure.getCurrentTest()
+
+        if (!this.isMultiremote) {
+            const { browserName, deviceName, desired, device } = this.capabilities
+            let targetName = device || browserName || deviceName || cid
+            // custom mobile grids can have device information in a `desired` cap
+            if (desired && desired.deviceName && desired.platformVersion) {
+                targetName = `${device || desired.deviceName} ${desired.platformVersion}`
+            }
+            const browserstackVersion = this.capabilities.os_version || this.capabilities.osVersion
+            const version = browserstackVersion || this.capabilities.browserVersion || this.capabilities.version || this.capabilities.platformVersion || ''
+            const paramName = (deviceName || device) ? 'device' : 'browser'
+            const paramValue = version ? `${targetName}-${version}` : targetName
+            currentTest.addParameter('argument', paramName, paramValue)
+        } else {
+            currentTest.addParameter('argument', 'isMultiremote', 'true')
+        }
+
+        // Allure analytics labels. See https://github.com/allure-framework/allure2/blob/master/Analytics.md
+        currentTest.addLabel('language', 'javascript')
+        currentTest.addLabel('framework', 'wdio')
+        currentTest.addLabel('thread', cid)
+    }
+
+    getLabels({
+        tags
+    }: SuiteStats) {
+        const labels: { name: string, value: string }[] = []
+        if (tags) {
+            (tags as Tag[]).forEach((tag: Tag) => {
+                const label = tag.name.replace(/[@]/, '').split('=')
+                if (label.length === 2) {
+                    labels.push({ name: label[0], value: label[1] })
+                }
+            })
+        }
+        return labels
+    }
+
+    onTestPass() {
+        if (this.options.useCucumberStepReporter) {
+            return this.allure.endStep('passed')
+        }
+
+        this.allure.endCase(PASSED)
+    }
+
+    onTestFail(test: TestStats | HookStats) {

```

Comment:
```suggestion
    onTestFail(test: TestStats) {

```

Reply:
Our code currently use this method for both tests and hooks, so we need to override type here